tests/test_bigbio.py

```diff
@@ -0,0 +1,660 @@
+
+        elif mapped_task == "text":
+            log.info("Sentence/Phrase/Text Classification Task")
+            self._check_keys(_SCHEMA_MAP[mapped_task], mapped_task)
+
+        elif mapped_task == "pairs":
+            log.info("Pair Labels Task")
+            self._check_keys(_SCHEMA_MAP[mapped_task], mapped_task)
+
+        else:
+            raise ValueError(
+                task
+                + " not recognized; only 'kb', 'qa', 'entailment', 'text_to_text', 'text', 'pairs'-type tasks are allowed"
+            )
+
+    def _test_task_exists(self, task_name: str):
+        """
+        Check if valid task name is provided
+        """
+        task = task_name.lower()
+        mapped_task = _TASK_MAP.get(task, None)
+        self.assertFalse(mapped_task is None, task + " is not recognized")
+        return mapped_task
+
+    @staticmethod
+    def _check_subkey(inp, attrs):
+        """Checks if subkeys (esp. in KB) have necessary criteria"""
+        return all([k in inp for k in attrs.keys()])
+
+    def _check_keys(self, schema: Features, schema_name: str):
+        """Check if necessary keys are present in a given schema"""
+        for split in self.dataset_bigbio.keys():
+            example = self.dataset_bigbio[split][0]
+
+            # Check for mandatory keys
+            mandatory_keys = all([key in example for key in schema.keys()])
+            self.assertTrue(
+                mandatory_keys,
+                "/".join(schema.keys())
+                + " keys missing from bigbio view for schema_type = "
+                + schema_name,
+            )
+
+    def _assert_ids_globally_unique(
+        self,
+        collection: Iterable,
+        ids_seen: set,
+        ignore_assertion: bool = False,
+    ):
+        """
+        Checks if all IDs are globally unique across a feature list.
+        This looks recursively through elements of arrays to check if every referenced ID is unique.
+
+        :param collection: An iterable of features that contain NLP info (ex: entities, events)
+        :param ids_seen: Set of previously seen numerical IDs (empty by default)
+        :param ignore_assertion: Whether to raise an error if id was already seen.
+        """  # noqa
+        if isinstance(collection, dict):
+
+            for k, v in collection.items():
+                if isinstance(v, dict):
+                    self._assert_ids_globally_unique(v, ids_seen)
+
+                elif isinstance(v, list):
+                    for elem in v:
+                        self._assert_ids_globally_unique(elem, ids_seen)
+                else:
+                    if k == "id":
+                        if not ignore_assertion:
+                            self.assertNotIn(v, ids_seen)
+                        ids_seen.add(v)
+
+        elif isinstance(collection, list):
+            for elem in collection:
+                self._assert_ids_globally_unique(elem, ids_seen)
```

Comment:
yea, maybe we can make this a warning? ... on the fence ... sometimes it will be a a dataset error, but often it will be an implementation error too ... definitely we should produce a warning for participants .. maybe we can sort the actual problem out during final pass Q/A.